pkg/config/types.go

```diff
@@ -163,7 +104,50 @@ func (t TestFixture) CfnTemplateFilename() string {
 	return t.cfnTemplateFilename
 }
 
-// AmiId returns amiId.
+// AmiId returns AmiId.
 func (t TestFixture) AmiId() string {
 	return t.amiId
 }
+
+// SetUserConfig sets empty fields of UserConfig to reqConfig
+func (UserConfig) SetUserConfig(reqConfig UserConfig) {
+	if userConfig.InstanceTypes == "" {
```

Comment:
Had a couple more if checks in here at first such as:

`if userConfig.InstanceTypes == "" && reqConfig.InstanceTypes != ""` , but go-report-card (mainly gocyclo) complained the whole function was too complex so had to limit to 15 if statements.

I think the right long-term move would be using Viper to handle the precedence for us, but I think this works for now